tsdate/core.py

```diff
@@ -940,6 +1136,8 @@ def date(
     Ne=None,
     return_posteriors=None,
     progress=False,
+    expectation_propagation=0,  # TODO document
```

Comment:
This serves two roles, to turn on the EP routine and also as the number of iterations, I think? Is it also reasonable to be able to iterate using the old ("inside outside") algorithm? In which case we should probably have 2 parameters, firstly `num_iterations`, and secondly `method`?

I'm also slightly unclear how this related to the `outside maximisation` method that previously existed. Would it be worth working out a scheme for how they all fit together (we can do this later, as long as the current routines are not released, or are undocumented)

Reply:
You mean, for a future version of tsdate where there's an interative algo for the old inside-outside method?

From the API side, maybe it's better to have a "control" argument that takes a dict of settings, that are used (or not) by whatever "method" is requested? Similar to what scipy does for its high level optimization functions. Then we won't have to translate every option across algorithms.

Reply:
Good point about a 'control' argument. I like that.